src/main_thread/decrypt/content_decryptor.ts

```diff
@@ -162,13 +178,14 @@ export default class ContentDecryptor extends EventEmitter<IContentDecryptorEven
       isInitDataQueueLocked: true,
       data: null,
     };
+    this._context = context;
     this.error = null;
 
     eme.onEncrypted(
       mediaElement,
       (evt) => {
         log.debug("DRM: Encrypted event received from media element.");
-        const initData = getInitData(evt as MediaEncryptedEvent);
+        const initData = getInitData(evt as ICustomMediaEncryptedEvent);
```

Comment:
We may want to better type `onEncrypted` instead?

Reply:
Is that still necessary if we update `onEncrypted`'s type in `IEmeApiImplementation` like this?:
```ts
  onEncrypted: (
    target: IEventTargetLike,
    listener: (evt: ICustomMediaEncryptedEvent) => void,
    cancelSignal: CancellationSignal,
  ) => void;

```

Reply:
Yes because `createCompatibleEventListener` returns a listener of type `(event?: Event) => void`
and will not satisfies type `(evt: ICustomMediaEncryptedEvent) => void`

Reply:
`Event` should be compatible to an `ICustomMediaEncryptedEvent` though as `forceSessionRecreation` is optional no?

Reply:
Yet maybe `Event` is not compatible with a `MediaEncryptedEvent`, in which case we might improve on the type?
Through `createCompatibleEventListener` event overloading?